pkg/reconciler/apibinding/apibinding_reconcile.go

```diff
@@ -330,48 +353,25 @@ func (r *workspaceAPIExportReferenceReconciler) reconcile(ctx context.Context, a
 
 			needToWaitForRequeue = true
 		} else {
-			// We never set Conversion, so nil it out before comparing
-			existingCRD = existingCRD.DeepCopy()
-			existingCRD.Spec.Conversion = nil
-
-			if !reflect.DeepEqual(existingCRD.Spec, crd.Spec) {
-				klog.Infof("CRD diff: %s", cmp.Diff(existingCRD.Spec, crd.Spec))
-				crd.ResourceVersion = existingCRD.ResourceVersion
-
-				if _, err := r.updateCRD(ctx, crd); err != nil {
-					if apierrors.IsInvalid(err) {
-						// this means that APIResourceSchema->CRD conversion failed
-						conditions.MarkFalse(
-							apiBinding,
-							apisv1alpha1.CRDReady,
-							apisv1alpha1.CreateErrorReason,
-							conditionsv1alpha1.ConditionSeverityError,
-							"error creating CRD %s: %v",
-							crd.Name,
-							err,
-						)
-						return reconcileStatusStop, nil // don't retry
-					}
-
-					return reconcileStatusStop, err // retry, temporary error
-				}
-
+			// Existing CRD flow
+			if !apihelpers.IsCRDConditionTrue(existingCRD, apiextensionsv1.Established) || apihelpers.IsCRDConditionTrue(existingCRD, apiextensionsv1.Terminating) {
 				needToWaitForRequeue = true
-			} else {
-				if !apihelpers.IsCRDConditionTrue(existingCRD, apiextensionsv1.Established) || apihelpers.IsCRDConditionTrue(existingCRD, apiextensionsv1.Terminating) {
-					needToWaitForRequeue = true
-				}
 			}
 		}
 
+		var storageVersions []string
+		if existingCRD != nil {
+			storageVersions = existingCRD.Status.StoredVersions
+		}
+
 		boundResources = append(boundResources, apisv1alpha1.BoundAPIResource{
 			Group:    schema.Spec.Group,
 			Resource: schema.Spec.Names.Plural,
 			Schema: apisv1alpha1.BoundAPIResourceSchema{
 				Name: schema.Name,
 				UID:  string(schema.UID),
 			},
-			StorageVersions: crd.Status.StoredVersions,
+			StorageVersions: storageVersions,
```

Comment:
When we switch schema a new bound CRD is created, right? Do we carry along storage versions?

Reply:
Yes, a new bound CRD would be created. I need to merge old+new storage versions.